drivers/sensor/bmm150/bmm150.c

```diff
@@ -623,12 +630,23 @@ static int bmm150_init_chip(const struct device *dev)
 	return -EIO;
 }
 
-#ifdef CONFIG_PM_DEVICE
 static int pm_action(const struct device *dev, enum pm_device_action action)
 {
-	int ret;
+	int ret = 0;
 
 	switch (action) {
+	case PM_DEVICE_ACTION_TURN_ON:
+		/* Re-Initialize Chip */
+		ret = bmm150_init_chip(dev);
+
+#ifdef CONFIG_BMM150_TRIGGER
+		if (bmm150_trigger_mode_init(dev) < 0) {
```

Comment:
Isn't this going to create the trigger thread each time the sensor is power cycled?

Reply:
true, I think I fixed that now.

Reply:
`trigger_mode_uninit` and `trigger_mode_init` are not quite balanced right ? There are many "one time thing" in `trigger_mode_init` that will end up being called multiple times now like `k_sem_init`, ` gpio_pin_configure_dt` ... 

Maybe it is worth to split this function and separate these static initialization from the pm logic.